themes/bootstrap3/templates/search/controls/showing.phtml

```diff
@@ -10,21 +10,25 @@
       isset($this->skipTotalCount) ? 'showing_results_html' : 'showing_results_of_html',
       $transParams
   );
-?>
-<?php // Use search-heading if it's set, but append only if it's not empty to avoid extra whitespace: ?>
-<?php if ($this->slot('search-heading')->isset()): ?>
-  <?php if ($heading = $this->slot('search-heading')->get()): ?>
-    <?php $showingResults .= " $heading"; ?>
-  <?php endif; ?>
-<?php elseif ($this->params->getSearchType() == 'basic'):
-  $showingResults = $this->translate(
-      isset($this->skipTotalCount) ? 'showing_results_for_html' : 'showing_results_of_for_html',
-      $transParams
-  );
-endif; ?>
 
+  // Use simple format if search-heading is set, search type is not basic or we have a simple top paginator enabled:
+  $topPaginator = $this->results ? $this->results->getOptions()->getTopPaginatorStyle() : '';
+  if ($this->slot('search-heading')->isset() || $this->params->getSearchType() !== 'basic' || 'simple' === $topPaginator) {
+    $showingResults = !empty($this->skipTotalCount)
```

Comment:
Isn't `$showingResults` _already_ set to this value at this point in the code due to lines 9-12 above? Since this if..else guarantees that the variable will be set to something, can we just remove those earlier lines?

Reply:
Oops, yes, though the condition wasn't quite the same! Now cleaned up.